pkg/streamingpromql/planning.go

```diff
@@ -75,6 +77,15 @@ type PlanningObserver interface {
 }
 
 func (p *QueryPlanner) NewQueryPlan(ctx context.Context, qs string, timeRange types.QueryTimeRange, observer PlanningObserver) (*planning.QueryPlan, error) {
+	if p.activeQueryTracker != nil {
```

Comment:
Don't feel obligated to change this but if we used a no-op version of the active query tracker when it's disabled we could avoid this nil checks.

Reply:
Good idea, added in 12d74e5.